### DISCUSSION TITLE
Should we always migrate the DB?

### DISCUSSION THREAD

**Discussion by fossbrandon** [jasontaylordev/CleanArchitecture#1256] [Q&A] [answered]
After looking through the template for the first time, it looks like the DB only runs migrations during development builds. Is there a reason this is the case? To me, it seems like the app should always initialize the DB when starting regardless of environment, but I am new to some of these topics so looking for clarification. Thanks!

**Comment by jasontaylordev** [comment, marked as answer]
The strategy mentioned initialises the database on application startup while preserving existing data. This is a productive and simple strategy for local development environments. However, it is not appropriate for other environments, such as staging and production. For these environments, database initialisation should occur during deployment.

https://jasontaylor.dev/ef-core-database-initialisation-strategies/
https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/applying?tabs=dotnet-core-cli#apply-migrations-at-runtime

**Reply by fossbrandon** [reply]
That's a great write up. Thank you for pointing me to it.
